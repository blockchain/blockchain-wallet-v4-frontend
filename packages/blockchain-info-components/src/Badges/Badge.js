import React from 'react'
import PropTypes from 'prop-types'

import { Image } from '../Images'
import { Link } from '../Links'

const Badge = ({ ...props }) => {
  const { size, type } = props
  switch (type) {
    case 'applestore':
      return (
        <Link
          href='https://itunes.apple.com/us/app/blockchain-bitcoin-wallet/id493253309'
          target='_blank'
        >
          <Image name='apple-app-store-badge' height={size || '48px'} />
        </Link>
      )
    case 'googleplay':
      return (
        <Link
          href='https://play.google.com/store/apps/details?id=piuk.blockchain.android'
          target='_blank'
        >
          <Image name='google-play-badge' height={size || '48px'} />
        </Link>
      )
    default:
      return <div />
  }
}

Badge.propTypes = {
  size: PropTypes.string,
  type: PropTypes.oneOf(['applestore', 'googleplay']).isRequired
}

export default Badge
