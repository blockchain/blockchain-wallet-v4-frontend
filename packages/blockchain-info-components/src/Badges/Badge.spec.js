import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'

import Badge from './Badge'

describe('Badge component', () => {
  it('applestore renders correctly', () => {
    const component = render(<Badge type='applestore' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })

  it('googleplay renders correctly', () => {
    const component = render(<Badge type='googleplay' />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
