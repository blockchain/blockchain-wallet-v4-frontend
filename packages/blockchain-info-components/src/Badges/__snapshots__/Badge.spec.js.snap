// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge component applestore renders correctly 1`] = `
<a
  class="Link__BaseLink-u5l9b-0 eiSGsv"
  color="brand-secondary"
  href="https://itunes.apple.com/us/app/blockchain-bitcoin-wallet/id493253309"
  target="_blank"
>
  <img
    class="Image__BaseImage-sc-1lfdauq-0 cpEzaa"
    color="auto"
    height="40px"
    src="[object Object]"
    srcset=""
    width="auto"
  />
</a>
`;

exports[`Badge component googleplay renders correctly 1`] = `
<a
  class="Link__BaseLink-u5l9b-0 eiSGsv"
  color="brand-secondary"
  href="https://play.google.com/store/apps/details?id=piuk.blockchain.android"
  target="_blank"
>
  <img
    class="Image__BaseImage-sc-1lfdauq-0 cpEzaa"
    color="auto"
    height="40px"
    src="[object Object]"
    srcset=""
    width="auto"
  />
</a>
`;
